Assets/MRTK/Core/Attributes/PreserveAttribute.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Microsoft.MixedReality.Toolkit
+{
+    /// <summary>
+    /// Custom preserve attribute that is inheritable.
+    /// </summary>
+    [AttributeUsage(
+        AttributeTargets.Assembly |
+        AttributeTargets.Class |
+        AttributeTargets.Struct |
+        AttributeTargets.Enum |
+        AttributeTargets.Constructor |
+        AttributeTargets.Method |
+        AttributeTargets.Property |
+        AttributeTargets.Field |
+        AttributeTargets.Event |
+        AttributeTargets.Interface |
+        AttributeTargets.Delegate,
+        Inherited = true)]
+    public class MixedRealityToolkitPreserveAttribute : UnityEngine.Scripting.PreserveAttribute
```

Comment:
I don't see this being used anywhere in this PR - I take it this is primarily for future use?

Can we add some more doccomments about when you should use this/what the purpose is etc?